### PR TITLE
Add read to `FitResult`

### PR DESCRIPTION
**Pull request by Astro-Kirsty**
For some cases the total_stat was saving as a `numpy.float64` which the yaml dump cannot understand
[See here](https://github.com/gammapy/gammapy/assets/60767360/d900ba2e-06fd-4e79-be9b-f1f3dde0743c)
Therefore we must ensure the `total_stat` is a `float` in the dictionary. 

For now I have a read function of the yaml. But it is unclear to me how we would like the output
Feel free to add your opinions here.

### CODE REVIEW

File: gammapy/modeling/tests/test_fit.py [hunk 1]
```diff
@@ -356,3 +356,11 @@ def test_write(tmpdir):
...
     assert "covariance_result" not in result_dict
     result.write(path=tmpdir / "test-fit-result.yaml", overwrite=True)
+
+    result_read = FitResult.read(filename=tmpdir / "test-fit-result.yaml")
+
+    assert result_read["optimize_result"]["nfev"] == result.optimize_result.nfev
+    assert (
+        result_read["optimize_result"]["total_stat"]
+        == result.optimize_result.total_stat
+    )
```
Comment (cgalelli) [inline gammapy/modeling/tests/test_fit.py:366]: I would just add a test catching the checksum fail here

**Review by cgalelli — COMMENTED** [review]
Thanks @Astro-Kirsty , as annotated I would maybe just add the test for the checksum verification fail.

File: gammapy/modeling/tests/test_fit.py (hunk 1 above)
Comment (Astro-Kirsty) [reply gammapy/modeling/tests/test_fit.py:366]: Thanks @cgalelli 

This actually needs to be updated after all the covariances changes have been added!

File: gammapy/modeling/fit.py [hunk 2]
```diff
@@ -825,6 +830,35 @@ def write(
...
+        data = yaml.safe_load(yaml_str)
+        checksum_str = data.pop("checksum", None)
+        if checksum:
+            yaml_str = yaml.dump(
+                data, sort_keys=False, indent=4, width=80, default_flow_style=False
+            )
+            if not verify_checksum(yaml_str, checksum_str):
+                warnings.warn("Checksum verification failed.", UserWarning)
+
+        return data
```
Comment (QRemy) [inline gammapy/modeling/fit.py:860]: This method is equivalent to `gammapy.utils.scripts.read_yaml`, so if the goal is just to  get a dict we don't need more.

Here .read and .to_yaml  returns a dict not an instance of the class so they are not classmethod.
The write method does not export all the mandatory arguments of OptimizeResult and CovarianceResult __init__ so we cannot really create a read method right now.

**Review by QRemy — COMMENTED** [review]
Thanks @Astro-Kirsty, I see several issues with this right now :

File: gammapy/modeling/fit.py (hunk 2 above)
Comment (QRemy) [reply gammapy/modeling/fit.py:860]: I would also say that with the covariance bug fixed in  #5317, the need to read back the output of FitResult is less obvious.

Comment (Astro-Kirsty) [reply gammapy/modeling/fit.py:860]: I agree with not really needing the read of the FitResult, now with all the other updates in the PRs have been made.

I suggest we close?

**Pull request closed without merging**